src/routes/loginOAuth.js

```diff
@@ -40,4 +40,43 @@ server.post("/auth/login", async (req, res) => {
   }
 });
 
+server.post("/auth/changePassword", async (req, res) => {
+  const { username, oldPassword, newPassword } = req.body;
+  let encryptedPassword
+  await encrypt.encryptPassword(newPassword).then(encryptedPass => {
+    encryptedPassword = encryptedPass;
+  })
+
+  const user = await Users.findOne({ username });
+  if (!user) {
+    res.status(401).json({
+      message: "Username not found",
+      error: "401: User not found",
+    });
+  } else {
+
+    if (user.password != undefined) {
```

Comment:
If we reach this part of the code, it means we found a user with the username / password pair so we don't need to check if the user has a password so we can get rid of that if statement also.
Other thing to mention here is that if you were still needed to do this check you could simply verify `if (user.password) ` without needing to specify `!= undefined`


```suggestion
```